pkg/logging/sftp/create.go

```diff
@@ -0,0 +1,154 @@
+	c.CmdClause.Flag("password", "The password for the server. If both password and secret_key are passed, secret_key will be used in preference").Action(c.Password.Set).StringVar(&c.Password.Value)
+	c.CmdClause.Flag("public-key", "A PGP public key that Fastly will use to encrypt your log files before writing them to disk").Action(c.PublicKey.Set).StringVar(&c.PublicKey.Value)
+	c.CmdClause.Flag("secret-key", "The SSH private key for the server. If both password and secret_key are passed, secret_key will be used in preference").Action(c.SecretKey.Set).StringVar(&c.SecretKey.Value)
+	c.CmdClause.Flag("path", "The path to upload logs to. The directory must exist on the SFTP server before logs can be saved to it").Action(c.Path.Set).StringVar(&c.Path.Value)
+	c.CmdClause.Flag("period", "How frequently log files are finalized so they can be available for reading (in seconds, default 3600)").Action(c.Period.Set).UintVar(&c.Period.Value)
+	c.CmdClause.Flag("format", "Apache style log formatting").Action(c.Format.Set).StringVar(&c.Format.Value)
+	c.CmdClause.Flag("format-version", "The version of the custom logging format used for the configured endpoint. Can be either 2 (default) or 1").Action(c.FormatVersion.Set).UintVar(&c.FormatVersion.Value)
+	c.CmdClause.Flag("gzip-level", "What level of GZIP encoding to have when dumping logs (default 0, no compression)").Action(c.GzipLevel.Set).UintVar(&c.GzipLevel.Value)
+	c.CmdClause.Flag("response-condition", "The name of an existing condition in the configured endpoint, or leave blank to always execute").Action(c.ResponseCondition.Set).StringVar(&c.ResponseCondition.Value)
+	c.CmdClause.Flag("timestamp-format", `strftime specified timestamp formatting (default "%Y-%m-%dT%H:%M:%S.000")`).Action(c.TimestampFormat.Set).StringVar(&c.TimestampFormat.Value)
+	c.CmdClause.Flag("placement", "Where in the generated VCL the logging call should be placed, overriding any format_version default. Can be none or waf_debug").Action(c.Placement.Set).StringVar(&c.Placement.Value)
+
+	return &c
+}
+
+// createInput transforms values parsed from CLI flags into an object to be used by the API client library.
+func (c *CreateCommand) createInput() (*fastly.CreateSFTPInput, error) {
+	var input fastly.CreateSFTPInput
+
+	serviceID, source := c.manifest.ServiceID()
+	if source == manifest.SourceUndefined {
+		return nil, errors.ErrNoServiceID
+	}
+
+	input.Service = serviceID
+	input.Version = c.Version
+	input.Name = fastly.String(c.EndpointName)
+	input.Address = fastly.String(c.Address)
+	input.User = fastly.String(c.User)
+	input.SSHKnownHosts = fastly.String(c.SSHKnownHosts)
+
+	if c.Port.Valid {
+		input.Port = fastly.Uint(c.Port.Value)
+	}
+
+	if c.Password.Valid {
+		input.Password = fastly.String(c.Password.Value)
+	}
+
+	if c.PublicKey.Valid {
+		input.PublicKey = fastly.String(c.PublicKey.Value)
+	}
+
+	if c.SecretKey.Valid {
+		input.SecretKey = fastly.String(c.SecretKey.Value)
+	}
```

Comment:
Thanks for pointing this out!

@phamann and I talked about this and decided to let validation be done by the API rather than having to be concerned about keeping validation consistent in many places.

Reply:
I have limited experience with the flag package that we use, namely [`alecthomas/kingpin`](https://pkg.go.dev/github.com/alecthomas/kingpin?tab=doc), but what would be ideal is if we could have "exclusive or" required fields. Otherwise, if we do want to enforce this validation in the CLI, I think we must do validation in this `createInput` method.